Library.java

```diff
@@ -0,0 +1,249 @@
+	}
+
+
+	public List<loan> getCurrentLoans() {
+		return new ArrayList<loan>(currentLoans.values());
+	}
+
+
+	public Member addMember(String lastName, String firstName, String email, int phoneNo) {		
+		Member member = new Member(lastName, firstName, email, phoneNo, getNextMemberID());
+		members.put(member.getID(), member);		
+		return member;
+	}
+
+	
+	public Book addBook(String author, String title, String callNumber) {		
+		Book book = new Book(author, title, callNumber, getNextBookID());
+		catalogue.put(book.getID(), book);		
+		return book;
+	}
+
+	
+	public Member getMember(int memberId) {
+		if (members.containsKey(memberId)){ 
+            return members.get(memberId);
+        }    
+		return null;
+	}
+
+	
+	public Book getBook(int bookId) {
+		if (catalogue.containsKey(bookId)) {
+            return catalogue.get(bookId);
+        }		
+		return null;
+	}
+
+	
+	public int getLoanLimit() {
+		return LOAN_LIMIT;
+	}
+
+	
+	public boolean approveLoan(Member member) {		
+        if (member.numberOfCurrentLoans() == LOAN_LIMIT ) {
+			return false;
+        }
+		if (member.finesOwed() >= MAX_FINES_OWED) {
+			return false;
+		}
+		for (loan loan : member.getLoans()) {
+			if (loan.OVer_Due()) {
+				return false;
+            }
+        }
+		return true;
+	}
+
+	
+	public int getRemainingLoans(Member member) {		
+		return LOAN_LIMIT - member.numberOfCurrentLoans();
+	}
+
+	
+	public loan issueLoan(Book book, Member member) {
+		Date dueDate = Calendar.getInstance().dueDate(LOAN_PERIOD);
+		loan loan = new loan(getNextLoanID(), book, member, dueDate);
+		member.takeOutLoan(loan);
+		book.borrowBook();
+		loans.put(loan.ID(), loan);
+		currentLoans.put(book.getID(), loan);
+		return loan;
+	}
+	
+	
+	public loan currentLoans(int bookID) {
+		if (currentLoans.containsKey(bookID)) {
+			return currentLoans.get(bookID);
+		}
+		return null;
+	}
+
+	
+	public double totalFinesOverdue(loan loan) {
+		if (loan.OVer_Due()) {
+            Date dueDate = loan.Get_Due_Date();
+			long daysOverDue = Calendar.getInstance().getDaysDifference(dueDate);
+			double fine = daysOverDue * FINE_PER_DAY;
+			return fine;
+		}
+		return 0.0;		
+	}
+
+
+	public void returnLoan(loan currentLoan, boolean isDamaged) {
+		Member member = currentLoan.Member();
+		Book book  = currentLoan.Book();
+		
+		double overDueFine = totalFinesOverdue(currentLoan);
+		member.addFine(overDueFine);	
+		
+		member.removeLoan(currentLoan);
+		book.returnBook(isDamaged);
+		if (isDamaged) {
+			member.addFine(DAMAGE_FEE);
+			damagedBooks.put(book.getID(), book);
+		}
+		currentLoan.DiScHaRgE();
+		currentLoans.remove(book.getID());
+	}
+
+
+	public void checkCurrentLoans() {
+		for (loan loan : currentLoans.values()) {
+			loan.checkOverDue();
+		}		
+	}
+
+
+	public void repairBook(Book currentBook) {
+		if (damagedBooks.containsKey(currentBook.getID())) {
+			currentBook.repairBook();
+			damagedBooks.remove(currentBook.getID());

```

Comment:
Please separate method within method call. Use the new bookID variable created above.
```suggestion
			damagedBooks.remove(bookID);
```